src/host.ps1

```diff
@@ -1,43 +1,97 @@
     [OutputType("XoPowershell.Host")]
     param (
-        [Parameter(Mandatory, ValueFromPipeline)]
-        [object]$InputObject
+        [Parameter(Mandatory, Position = 0)]
+        $InputObject
+    )
+
+    $hostObj = [PSCustomObject]@{
+        PSTypeName    = "XoPowershell.Host"
+        HostUuid      = $InputObject.uuid
+        Name          = $InputObject.name_label
+        Address       = $InputObject.address
+        PowerState    = $InputObject.power_state
+        Description   = $InputObject.name_description
+        StartTime     = $InputObject.startTime
+        Tags          = $InputObject.tags
+        Version       = $InputObject.version
+        ProductBrand  = $InputObject.productBrand
+        BiosStrings   = $InputObject.bios_strings
+        Build         = $InputObject.build
+        Hostname      = $InputObject.hostname
+        LicenseParams = $InputObject.license_params
+        LicenseServer = $InputObject.license_server
+        LicenseExpiry = $InputObject.license_expiry
+        ResidentVms   = $InputObject.residentVms
+        Pifs          = $InputObject.PIFs
+        PcIs          = $InputObject.PCIs
+        PGpus         = $InputObject.PGPUs
+        PoolId        = $InputObject.poolId
+        Memory        = $InputObject.memory
+        CPUs          = $InputObject.CPUs
+    }
+
+    if ($InputObject.CPUs -and $InputObject.CPUs.cpu_count) {
+        $hostObj | Add-Member -NotePropertyName "VCpus" -NotePropertyValue $InputObject.CPUs.cpu_count
+    } elseif ($InputObject.cpus -and $InputObject.cpus.cores) {
+        $hostObj | Add-Member -NotePropertyName "VCpus" -NotePropertyValue $InputObject.cpus.cores
+    }
+    
+    return $hostObj
+}
+
+function Get-XoSingleHostById {
+    param (
+        [string]$HostId,
+        [hashtable]$Params
     )
     
-    process {
-        $resultObject = [PSCustomObject]@{
-            PSTypeName = "XoPowershell.Host"
-            HostUuid = $InputObject.id
-            Description = $InputObject.name_description
-            Label = $InputObject.name_label
-            MemoryFree = [int64]$InputObject.memory.free
-            MemoryTotal = [int64]$InputObject.memory.total
-            Cpus = $InputObject.cpus.cores
-            Power = $InputObject.power_state
-            StartTime = $InputObject.startTime
-            Tags = $InputObject.tags
-            Address = $InputObject.address
-            Bios = $InputObject.bios_strings.bios_vendor
-            CPUModel = $InputObject.cpu_info.modelname
-            Build = $InputObject.software_version.build_number
-            Version = $InputObject.software_version.product_version
-            ProductBrand = $InputObject.software_version.product_brand
+    try {
+        $uri = "$script:XoHost/rest/v0/hosts/$HostId"
+        $hostData = Invoke-RestMethod -Uri $uri @script:XoRestParameters -Body ($Params | ConvertTo-Json -Compress) -Method Get
+        
+        $hostData = $hostData | ConvertFrom-Json -AsHashTable
```

Comment:
`Invoke-RestMethod` already returns a parsed JSON object, why the conversion? Also, I'm pretty sure that's not the right way to pass parameters to the JSON request.